controller/service.go

```diff
@@ -118,10 +127,19 @@ func (c *controller) convergeBalancer(l log.Logger, key string, svc *v1.Service)
 			c.client.Errorf(svc, "LoadBalancerFailed", "invalid requested loadbalancer IPs: %s", err)
 			return true
 		}
-		if len(desiredLbIPs) > 0 && !isEqualIPs(lbIPs, desiredLbIPs) {
-			level.Info(l).Log("event", "clearAssignment", "reason", "differentIPRequested", "msg", "user requested a different IP than the one currently assigned")
-			c.clearServiceState(key, svc)
-			lbIPs = []net.IP{}
+		if len(desiredLbIPs) > 0 {
```

Comment:
Can we split this check in two, with one being the old one (the user requested a different set of ips). I think it will be eeasier to follow as the logic will be isolated.
So, two checks:
- user requested different ip
- user requested ips from a pool he has no access to